dirty_cat/super_vectorizer.py

```diff
@@ -208,7 +198,30 @@ def _auto_cast(X: pd.DataFrame) -> pd.DataFrame:
                 if pd.api.types.is_numeric_dtype(X[col]):
                     X[col] = X[col].astype(np.float64)
                 X[col].fillna(value=np.nan, inplace=True)
+        STR_NA_VALUES = ['null', '', '1.#QNAN', '#NA', 'nan', '#N/A N/A', '-1.#QNAN', '<NA>', '-1.#IND', '-nan', 'n/a',
+                         '-NaN', '1.#IND', 'NULL', 'NA', 'N/A', '#N/A', 'NaN']  # taken from pandas.io.parsers (version 1.1.4)
+        X = X.replace(STR_NA_VALUES + [None, "?", "..."],
```

Comment:
I'm surprised that this works if X is a numpy array. I actually think that it doesn't work.

It seems that we are indeed testing properly the SuperVectorizer with numpy arrays. I'm a bit confused :)